src/packet_analysis/Analyzer.cc

```diff
@@ -47,14 +47,24 @@ bool Analyzer::RegisterAnalyzerMapping(uint32_t identifier, AnalyzerPtr analyzer
 	return dispatcher.Register(identifier, std::move(analyzer));
 	}
 
+void Analyzer::RegisterDefaultAnalyzer(AnalyzerPtr default_analyzer)
+	{
+	this->default_analyzer = std::move(default_analyzer);
+	}
+
+AnalyzerPtr Analyzer::Lookup(uint32_t identifier) const
+	{
+	return dispatcher.Lookup(identifier);
+	}
+
 AnalyzerResult Analyzer::AnalyzeInnerPacket(Packet* packet,
 		const uint8_t*& data, uint32_t identifier) const
 	{
-	auto inner_analyzer = dispatcher.Lookup(identifier);
+	auto inner_analyzer = Lookup(identifier);
+	inner_analyzer = inner_analyzer ? inner_analyzer : default_analyzer;
```

Comment:
I know it gives the same results, but I feel like the below is a bit clearer. That said, I wonder what the compiler actually does there.

```suggestion
	if ( ! inner_analyzer )
		inner_analyzer = default_analyzer;
```

Reply:
I can see that the ``if`` looks cleaner and changed it.